core/session.ts

```diff
@@ -191,6 +196,7 @@ export class Session {
           ? this.config.assertionSchema
           : this.config.defaultSchema);
       action.proto.target = this.target(actionOptions.sqlxConfig.name, finalSchema);
+      action.proto.name = action.proto.target.schema + "." + action.proto.target.name;
```

Comment:
after this PR, `name` and `target` are equivalently useable for uniqueness. I think we should:
1. put a comment on the `name` field to indicate that it is deprecated, and to prefer using the `target` field
2. stop unsetting the `target` field in the case of `Operation` queries
3. (as a direct result of (2)) change `resolve` to throw if you try to resolve an `Operation` with `hasOutput = false`

Reply:
Done. Notice that as a corollary of this, the following test will be changed:
https://github.com/dataform-co/dataform/blob/58a5226eb5a8b57ae0e16bdd1852fb41a2ae6736/tests/api/api.spec.ts#L824